docs/rfcs/016-holdings_service/README.md

```diff
@@ -0,0 +1,128 @@
+# **RFC 016: Holdings service**
+
+**Last updated: 23 September 2019.**
+
+## **Background**
+
+As we move a greater proportion of people over from wellcomelibrary.org to
+wellcomecollection.org we need to provide a service allowing people to query on
+and request items based on their availability and
+location, whether it be online, in the library, or something else.
+
+This document will outline the different services we will need build and
+integrate with.
+
+We will explore the best way to start developing the service and get it to
+people for research and testing as quickly as possible, without losing sight of
+the fact that this service will need to integrate with Sierra, CAS and other
+services while containing other integral pieces of infrastructure.
+
+We will also assume that we will not be disrupting any current processes across
+teams outside of the digital platform, or adding more places for library members
+to manage their relationship with us (i.e. account information).
+
+The project will most likely be a cross team (experience / data / preservation)
+endeavour, so the process in how we communicate and share what we are doing will
+be important too.
+
+## **Proposed solution**
+
+[https://docs.google.com/drawings/d/1jAAiUS2A2wifiV2ZL5RkdPzHDHvOzTSQPE-HWwxeyMU/edit?usp=sharing](https://docs.google.com/drawings/d/1jAAiUS2A2wifiV2ZL5RkdPzHDHvOzTSQPE-HWwxeyMU/edit?usp=sharing)
+
+### Client (Experience)
+
+At first the client would make calls to a separate Holdings API and
+progressively enhance the page as we would not want to block rendering of the
+integral catalogue information, nor could we rely on the consistency of the data
+with Sierra, at least at first. See data principles below.
+
+### Identity (Preservation)
+
+Requests will need to be authenticated and identified through the current
+identity service, CAS, as we will need to update Sierra with requests to not
+disrupt the current workflow of library desk staff.
+
+This authentication can happen as the first step and hopefully we will be able
+to propegate a token through to the holdings API which will in turn us it to
+authenticate with the sierra integration.
+
+There is currently a process for setting up a new service with SSO where the
+keys and knowledge sit with Digirati.
+
+We need to make sure we don't get sidelined into recreating a whole identity
+management system as that is not in scope, but to integrate with what we have.
+
+**Questions**
+
+- Do we need to have a proxy API in front of this?
+- Do we just keep using the library application already setup?
+
+### Holdings API (Data)
```

Comment:
Can we have a go at some endpoints here? This probably needs describing in Swagger as a first pass.